h/static/scripts/annotator/guest.coffee

```diff
@@ -85,100 +86,49 @@ module.exports = class Guest extends Annotator
-
-    # Watch for newly rendered highlights, and update positions in sidebar
-    this.subscribe "highlightsCreated", (highlights) =>
-      unless Array.isArray highlights
-        highlights = [highlights]
-      highlights.forEach (hl) ->
-        hls = hl.anchor.highlight  # Fetch all the highlights
-        # Get the pages we have highlights on (for the given anchor)
-        pages = Object.keys(hls).map (s) -> parseInt s
-        firstPage = pages.sort()[0]  # Determine the first page
-        firstHl = hls[firstPage]     # Determine the first (topmost) hl
-        # Store the position of this anchor inside target
-        hl.anchor.target.pos =
-          top: hl.getTop()
-          height: hl.getHeight()
-
-      # Collect all impacted annotations
-      annotations = (hl.annotation for hl in highlights)
-
-      # Announce the new positions, so that the sidebar knows
-      this.plugins.CrossFrame.sync(annotations)
-
-    # Watch for removed highlights, and update positions in sidebar
-    this.subscribe "highlightRemoved", (highlight) =>
-      hls = highlight.anchor.highlight  # Fetch all the highlights
-      # Get the pages we have highlights on (for the given anchor)
-      pages = Object.keys(hls).map (s) -> parseInt s
-      # Do we have any highlights left?
-      if pages.length
-        firstPage = pages.sort()[0]  # Determine the first page
-        firstHl = hls[firstPage]     # Determine the first (topmost) hl
-        # Store the position of this anchor inside target
-        highlight.anchor.target.pos =
-          top: highlight.getTop()
-          heigth: highlight.getHeight()
-      else
-        delete highlight.anchor.target.pos
-
-      # Announce the new positions, so that the sidebar knows
-      this.plugins.CrossFrame.sync([highlight.annotation])
-
-  # Utility function to remove the hash part from a URL
-  _removeHash: (url) ->
-    url = new URL url
-    url.hash = ""
-    url.toString()
-
-  # Utility function to get the decoded form of the document URI
-  getRawHref: ->
-    if @plugins.PDF
-      @plugins.PDF.uri()
+  # Get the document info
+  getDocumentInfo: ->
+    if @plugins.PDF?
+      metadataPromise = Promise.resolve(@plugins.PDF.getMetadata())
+      uriPromise = Promise.resolve(@plugins.PDF.uri())
+    else if @plugins.Document?
+      uriPromise = Promise.resolve(@plugins.Document.uri())
+      metadataPromise = Promise.resolve(@plugins.Document.metadata)
     else
-      @plugins.Document.uri()
-
-  # Utility function to get a de-hashed form of the document URI
-  getHref: -> @_removeHash @getRawHref()
-
-  # Utility function to filter metadata and de-hash the URIs
-  getMetadata: =>
-    metadata = @plugins.Document?.metadata
-    metadata.link?.forEach (link) => link.href = @_removeHash link.href
-    metadata
+      uriPromise = Promise.reject()
+      metadataPromise = Promise.reject()
+
+    uriPromise = uriPromise.catch(-> decodeURIComponent(window.location.href))
```

Comment:
Fallsbacks. There's a sensible result even without any plugin support for document metadata.